recipes/pybind11/all/conanfile.py

```diff
@@ -50,24 +43,39 @@ def package(self):
         self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
         cmake = self._configure_cmake()
         cmake.install()
-        lib_folder = os.path.join(
-            self.package_folder, "lib", "cmake", "pybind11")
-        os.rename(
-            os.path.join(lib_folder, "pybind11Config.cmake"),
-            os.path.join(lib_folder, "pybind11Install.cmake"))
-        os.unlink(os.path.join(lib_folder, "pybind11ConfigVersion.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Common.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11NewTools.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Targets.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11Config.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "pybind11", "pybind11ConfigVersion.cmake"))
 
     def package_id(self):
         self.info.header_only()
 
     def package_info(self):
-        self.cpp_info.includedirs.append(os.path.join(
-            self.package_folder, "include", "pybind11"))
-
         cmake_base_path = os.path.join("lib", "cmake", "pybind11")
-        self.cpp_info.builddirs = [cmake_base_path]
+        if tools.Version(self.version) >= "2.6.0":
+            self.cpp_info.components["main"].names["cmake_find_package"] = "pybind11"
+            self.cpp_info.components["main"].builddirs = [cmake_base_path]
+            for generator in ["cmake_find_package", "cmake_find_package_multi"]:
+                self.cpp_info.components["main"].build_modules[generator].append(os.path.join(cmake_base_path, "FindPythonLibsNew.cmake"))
```

Comment:
Including `FindPythonLibsNew` will enforce use of the deprecated `FindPythonInterp` etc. pybind supports the new `FindPython` modules (CMake 3.12+) but if you remove `pybind11NewTools` etc, it won't be able to use it.

Maybe make it an option? See also #4641

Reply:
Recent work on this PR may address the issue described in #4641. Please review it if possible and consider closing the #4641 PR if that is the case. Thank you!